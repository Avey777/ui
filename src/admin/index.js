import UsersPage from './containers/UsersPage';
export {UsersPage};
