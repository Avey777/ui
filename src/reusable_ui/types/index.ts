export enum ComponentColor {
  Default = 'default',
  Primary = 'primary',
  Success = 'success',
  Warning = 'warning',
  Danger = 'danger',
  Alert = 'alert',
}

export enum ComponentSize {
  ExtraSmall = 'xs',
  Small = 'sm',
  Medium = 'md',
  Large = 'lg',
}

export enum ButtonShape {
  Default = 'none',
  Square = 'square',
  StretchToFit = 'stretch',
}
