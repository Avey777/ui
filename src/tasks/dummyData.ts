import {Task, TaskStatus} from 'src/types/v2/tasks'

export const dummyTasks: Task[] = [
  {
    id: '02ef9deff2141000',
    organizationId: '02ee9e2a29d73000',
    name: 'pasdlak',
    status: TaskStatus.Active,
    owner: {id: '02ee9e2a19d73000', name: ''},
    flux:
      'option task = {\n  name: "pasdlak",\n  cron: "2 0 * * *"\n}\nfrom(bucket: "inbucket") \n|> range(start: -1h)',
    cron: '2 0 * * *',
    organization: {
      links: {
        buckets: '/api/v2/buckets?org=RadicalOrganization',
        dashboards: '/api/v2/dashboards?org=RadicalOrganization',
        log: '/api/v2/orgs/02ee9e2a29d73000/log',
        members: '/api/v2/orgs/02ee9e2a29d73000/members',
        self: '/api/v2/orgs/02ee9e2a29d73000',
        tasks: '/api/v2/tasks?org=RadicalOrganization',
      },
      id: '02ee9e2a29d73000',
      name: 'RadicalOrganization',
    },
  },
  {
    id: '02f12c50dba72000',
    organizationId: '02ee9e2a29d73000',
    name: 'somename',
    status: TaskStatus.Active,
    owner: {id: '02ee9e2a19d73000', name: ''},
    flux:
      'option task = {\n  name: "somename",\n  every: 1m,\n}\nfrom(bucket: "inbucket") \n|> range(start: -task.every)',
    every: '1m0s',
    organization: {
      links: {
        buckets: '/api/v2/buckets?org=RadicalOrganization',
        dashboards: '/api/v2/dashboards?org=RadicalOrganization',
        log: '/api/v2/orgs/02ee9e2a29d73000/log',
        members: '/api/v2/orgs/02ee9e2a29d73000/members',
        self: '/api/v2/orgs/02ee9e2a29d73000',
        tasks: '/api/v2/tasks?org=RadicalOrganization',
      },
      id: '02ee9e2a29d73000',
      name: 'RadicalOrganization',
    },
  },
]
