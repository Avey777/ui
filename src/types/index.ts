import {AuthLinks, Organization, Role, User, Me} from './auth'
import {AlertRule, Kapacitor} from './kapacitor'
import {Query, QueryConfig} from './query'
import {Source} from './sources'
import {DropdownAction, DropdownItem} from './shared'

export {
  Me,
  AuthLinks,
  Role,
  User,
  Organization,
  AlertRule,
  Kapacitor,
  Query,
  QueryConfig,
  Source,
  DropdownAction,
  DropdownItem,
}
