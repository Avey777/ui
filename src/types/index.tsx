import {AuthLinks, Role, User, Organization} from './auth'
import {AlertRule, Kapacitor} from "./kapacitor"
import {Query, QueryConfig} from "./query"
import {Source} from "./sources"

export {AuthLinks, Role, User, Organization, AlertRule, Kapacitor, Query, QueryConfig, Source, }
