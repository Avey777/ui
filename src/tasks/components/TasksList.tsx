// Libraries
import React, {PureComponent} from 'react'
import {connect} from 'react-redux'
import _ from 'lodash'

// Components
import {ResourceList} from 'src/clockface'
import TaskCard from 'src/tasks/components/TaskCard'

// Types
import EmptyTasksList from 'src/tasks/components/EmptyTasksList'
import {ITask as Task} from '@influxdata/influx'
import {SortTypes} from 'src/shared/selectors/sort'
import {AppState} from 'src/types'
import {Sort} from '@influxdata/clockface'

import {
  addTaskLabelsAsync,
  removeTaskLabelsAsync,
  runTask,
} from 'src/tasks/actions'

// Selectors
import {getSortedResource} from 'src/shared/selectors/sort'

interface OwnProps {
  tasks: Task[]
  searchTerm: string
  onActivate: (task: Task) => void
  onDelete: (task: Task) => void
  onCreate: () => void
  onSelect: (task: Task) => void
  onClone: (task: Task) => void
  onFilterChange: (searchTerm: string) => void
  totalCount: number
  onRemoveTaskLabels: typeof removeTaskLabelsAsync
  onAddTaskLabels: typeof addTaskLabelsAsync
  onRunTask: typeof runTask
  onUpdate: (task: Task) => void
  filterComponent?: () => JSX.Element
  onImportTask: () => void
  sortKey: string
  sortDirection: Sort
  sortType: SortTypes
  onClickColumn: (nextSort: Sort, sortKey: SortKey) => void
}

interface StateProps {
  sortedTasks: Task[]
}

type Props = OwnProps & StateProps

type SortKey = keyof Task

interface State {
  taskLabelsEdit: Task
  isEditingTaskLabels: boolean
  sortedTasks: Task[]
}

class TasksList extends PureComponent<Props, State> {
  constructor(props) {
    super(props)
    this.state = {
      taskLabelsEdit: null,
      isEditingTaskLabels: false,
      sortedTasks: this.props.sortedTasks,
    }
  }

  componentDidUpdate(prevProps) {
    const {tasks, sortedTasks, sortKey, sortDirection} = this.props
    if (
      prevProps.sortDirection !== sortDirection ||
      prevProps.sortKey !== sortKey ||
      prevProps.tasks.length !== tasks.length
    ) {
      this.setState({sortedTasks})
    }
  }

  public render() {
    const {
      searchTerm,
      onCreate,
      totalCount,
      filterComponent,
      onImportTask,
      sortKey,
      sortDirection,
      onClickColumn,
    } = this.props

    const headerKeys: SortKey[] = ['name', 'status', 'every', 'latestCompleted']

    return (
      <>
        <ResourceList>
          <ResourceList.Header filterComponent={filterComponent}>
            <ResourceList.Sorter
              name="Name"
              sortKey={headerKeys[0]}
              sort={sortKey === headerKeys[0] ? sortDirection : Sort.None}
              onClick={onClickColumn}
            />
            <ResourceList.Sorter
              name="Active"
              sortKey={headerKeys[1]}
              sort={sortKey === headerKeys[1] ? sortDirection : Sort.None}
              onClick={onClickColumn}
            />
            <ResourceList.Sorter
              name="Schedule"
              sortKey={headerKeys[2]}
              sort={sortKey === headerKeys[2] ? sortDirection : Sort.None}
              onClick={onClickColumn}
            />
            <ResourceList.Sorter
              name="Last Completed"
              sortKey={headerKeys[3]}
              sort={sortKey === headerKeys[3] ? sortDirection : Sort.None}
              onClick={onClickColumn}
            />
          </ResourceList.Header>
          <ResourceList.Body
            emptyState={
              <EmptyTasksList
                searchTerm={searchTerm}
                onCreate={onCreate}
                totalCount={totalCount}
                onImportTask={onImportTask}
              />
            }
          >
            {this.rows}
          </ResourceList.Body>
        </ResourceList>
      </>
    )
  }

  private get rows(): JSX.Element[] {
    const {
      onActivate,
      onDelete,
      onSelect,
      onClone,
      onUpdate,
      onRunTask,
      onFilterChange,
    } = this.props
    const {sortedTasks} = this.state

    return sortedTasks.map(task => (
      <TaskCard
        key={`task-id--${task.id}`}
        task={task}
        onActivate={onActivate}
        onDelete={onDelete}
        onClone={onClone}
        onSelect={onSelect}
        onUpdate={onUpdate}
        onRunTask={onRunTask}
        onFilterChange={onFilterChange}
      />
    ))
  }
}

const mstp = (state: AppState, props: OwnProps): StateProps => {
  return {
    sortedTasks: getSortedResource(state.tasks.list, props),
  }
}

export default connect<StateProps, {}, OwnProps>(mstp)(TasksList)
