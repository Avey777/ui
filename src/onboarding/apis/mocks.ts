import {telegrafConfigsResponse, authResponse} from 'mocks/dummyData'

const telegrafsGet = jest.fn(() => Promise.resolve(telegrafConfigsResponse))
const authorizationsGet = jest.fn(() => Promise.resolve(authResponse))

export const telegrafsAPI = {
  telegrafsGet,
}

export const authorizationsAPI = {
  authorizationsGet,
}
