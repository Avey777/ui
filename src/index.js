import React from 'react'
import {render} from 'react-dom'
import {Provider} from 'react-redux'
import {Router, Route, useRouterHistory} from 'react-router'
import {createHistory} from 'history'
import {syncHistoryWithStore} from 'react-router-redux'

import App from 'src/App'
import AlertsApp from 'src/alerts'
import CheckSources from 'src/CheckSources'
import {HostsPage, HostPage} from 'src/hosts'
import {KubernetesPage} from 'src/kubernetes'
import {Login, UserIsAuthenticated, Authenticated, UserIsNotAuthenticated} from 'src/auth'
import {KapacitorPage, KapacitorRulePage, KapacitorRulesPage, KapacitorTasksPage} from 'src/kapacitor'
import DataExplorer from 'src/data_explorer'
import {DashboardsPage, DashboardPage} from 'src/dashboards'
import {CreateSource, SourcePage, ManageSources} from 'src/sources'
import {AdminPage} from 'src/admin'
import NotFound from 'src/shared/components/NotFound'
import configureStore from 'src/store/configureStore'
import {getMe, getSources} from 'shared/apis'
import {authRequested, authReceived, meRequested, meReceived} from 'shared/actions/auth'
import {disablePresentationMode} from 'shared/actions/app'
import {publishNotification} from 'shared/actions/notifications'
import {loadLocalStorage} from './localStorage'

import 'src/style/chronograf.scss'

import {HTTP_FORBIDDEN, HEARTBEAT_INTERVAL} from 'shared/constants'

const rootNode = document.getElementById('react-root')

let browserHistory
const basepath = rootNode.dataset.basepath
window.basepath = basepath
if (basepath) {
  browserHistory = useRouterHistory(createHistory)({
    basename: basepath, // this is written in when available by the URL prefixer middleware
  })
} else {
  browserHistory = useRouterHistory(createHistory)({
    basename: "",
  })
}

const store = configureStore(loadLocalStorage(), browserHistory)
const {dispatch} = store

browserHistory.listen(() => {
  dispatch(disablePresentationMode())
})

window.addEventListener('keyup', (event) => {
  if (event.key === 'Escape') {
    dispatch(disablePresentationMode())
  }
})

const history = syncHistoryWithStore(browserHistory, store)

const Root = React.createClass({
  getInitialState() {
    return {
      hasDeterminedAuth: false,
    }
  },

  componentWillMount() {
    this.checkAuth()
  },

  activeSource(sources) {
    const defaultSource = sources.find((s) => s.default)
    if (defaultSource && defaultSource.id) {
      return defaultSource
    }
    return sources[0]
  },

  async checkAuth() {
    dispatch(authRequested())
    dispatch(meRequested())
    try {
      await this.startHeartbeat({shouldDispatchResponse: true})
    } catch (error) {
      console.error(error)
    }
  },

  // TODO: use once auth router is working
  redirectFromRoot(_, replace, callback) {
    getSources().then(({data: {sources}}) => {
      if (sources && sources.length) {
        const path = `/sources/${this.activeSource(sources).id}/hosts`
        replace(path)
      }
      callback()
    })
  },

  async startHeartbeat({shouldDispatchResponse}) {
    try {
      const {data: me, auth} = await getMe()
      if (shouldDispatchResponse) {
        dispatch(authReceived(auth))
        dispatch(meReceived(me))
      }

      setTimeout(this.startHeartbeat.bind(null, {shouldDispatchResponse: false}), HEARTBEAT_INTERVAL)
    } catch (error) {
      if (error.auth) {
        dispatch(authReceived(error.auth))
        dispatch(meReceived(null))
      }
      if (error.status === HTTP_FORBIDDEN) {
        dispatch(publishNotification('error', 'Session timed out. Please login again.'))
      } else {
        dispatch(publishNotification('error', 'Cannot communicate with server.'))
      }
    }
  },

  render() {
    return (
      <Provider store={store}>
        <Router history={history}>
          <Route path="/" component={UserIsAuthenticated(CreateSource)/* add back onEnter={this.redirectFromRoot} */} />
          <Route path="login" component={UserIsNotAuthenticated(Login)} />
          <Route path="sources/new" component={UserIsAuthenticated(CreateSource)} />
          <Route path="sources/:sourceID" component={UserIsAuthenticated(App)}>
            <Route component={UserIsAuthenticated(CheckSources)}>
              <Route component={Authenticated}>
                <Route path="manage-sources" component={ManageSources} />
                <Route path="manage-sources/new" component={SourcePage} />
                <Route path="manage-sources/:id/edit" component={SourcePage} />
                <Route path="chronograf/data-explorer" component={DataExplorer} />
                <Route path="hosts" component={HostsPage} />
                <Route path="hosts/:hostID" component={HostPage} />
                <Route path="kubernetes" component={KubernetesPage} />
                <Route path="kapacitor-config" component={KapacitorPage} />
                <Route path="kapacitor-tasks" component={KapacitorTasksPage} />
                <Route path="alerts" component={AlertsApp} />
                <Route path="dashboards" component={DashboardsPage} />
                <Route path="dashboards/:dashboardID" component={DashboardPage} />
                <Route path="alert-rules" component={KapacitorRulesPage} />
                <Route path="alert-rules/:ruleID" component={KapacitorRulePage} />
                <Route path="alert-rules/new" component={KapacitorRulePage} />
                <Route path="admin" component={AdminPage} />
              </Route>
            </Route>
          </Route>
          <Route path="*" component={NotFound} />
        </Router>
      </Provider>
    )
  },
})

if (rootNode) {
  render(<Root />, rootNode)
}
