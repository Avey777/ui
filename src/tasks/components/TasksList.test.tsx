// Libraries
import React from 'react'
import {renderWithRedux} from 'src/mockState'

// Components
import TasksList from 'src/tasks/components/TasksList'

// Types
import {Task} from 'src/types'

// Constants
import {tasks} from 'mocks/dummyData'

const setup = (override?) => {
  const props = {
    tasks,
    searchTerm: '',
    onActivate: oneTestFunction,
    onDelete: oneTestFunction,
    onCreate: secondTestFunction,
    onSelect: oneTestFunction,
    onImportTask: oneTestFunction,
    ...override,
  }

  const wrapper = renderWithRedux(<TasksList {...props} />)

  return {wrapper}
}

const oneTestFunction = (tasks: Task) => {
  tasks[0].name = 'someone'
  return
}

const secondTestFunction = () => {
  return
}

describe('TasksList', () => {
  describe('rendering', () => {
    it('renders', () => {
      const {wrapper} = setup()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
