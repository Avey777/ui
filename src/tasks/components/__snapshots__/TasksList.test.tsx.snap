// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksList rendering renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="resource-list"
      >
        <div
          class="resource-list--header"
        >
          <div
            class="resource-list--filter"
          />
          <div
            class="resource-list--sorting"
          >
            <div
              class="resource-list--sorter"
              title="Sort Name in asc order"
            >
              Name
            </div>
            <div
              class="resource-list--sorter"
              title="Sort Active in asc order"
            >
              Active
            </div>
            <div
              class="resource-list--sorter"
              title="Sort Schedule in asc order"
            >
              Schedule
            </div>
            <div
              class="resource-list--sorter"
              title="Sort Last Completed in asc order"
            >
              Last Completed
            </div>
          </div>
        </div>
        <div
          class="resource-list--body"
        >
          <div
            class="empty-state empty-state--lg"
            data-testid="empty-state"
          >
            <h4
              class="empty-state--text"
              data-testid="empty-state--text"
            >
              Looks 
              like 
              you 
              don't 
              have 
              any 
              <em>
                Tasks
              </em>
              , 
              why 
              not 
              create 
              one? 
            </h4>
            <div
              class="dropdown dropdown-sm dropdown-primary dropdown-truncate dropdown--action"
              style="width: 160px;"
            >
              <button
                class="dropdown--button button button-stretch button-primary button-sm"
                data-testid="dropdown-button"
                title="Create Task"
                type="button"
              >
                <span
                  class="dropdown--icon icon plus"
                />
                <span
                  class="dropdown--selected"
                >
                  Create Task
                </span>
                <span
                  class="dropdown--caret icon caret-down"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="resource-list"
    >
      <div
        class="resource-list--header"
      >
        <div
          class="resource-list--filter"
        />
        <div
          class="resource-list--sorting"
        >
          <div
            class="resource-list--sorter"
            title="Sort Name in asc order"
          >
            Name
          </div>
          <div
            class="resource-list--sorter"
            title="Sort Active in asc order"
          >
            Active
          </div>
          <div
            class="resource-list--sorter"
            title="Sort Schedule in asc order"
          >
            Schedule
          </div>
          <div
            class="resource-list--sorter"
            title="Sort Last Completed in asc order"
          >
            Last Completed
          </div>
        </div>
      </div>
      <div
        class="resource-list--body"
      >
        <div
          class="empty-state empty-state--lg"
          data-testid="empty-state"
        >
          <h4
            class="empty-state--text"
            data-testid="empty-state--text"
          >
            Looks 
            like 
            you 
            don't 
            have 
            any 
            <em>
              Tasks
            </em>
            , 
            why 
            not 
            create 
            one? 
          </h4>
          <div
            class="dropdown dropdown-sm dropdown-primary dropdown-truncate dropdown--action"
            style="width: 160px;"
          >
            <button
              class="dropdown--button button button-stretch button-primary button-sm"
              data-testid="dropdown-button"
              title="Create Task"
              type="button"
            >
              <span
                class="dropdown--icon icon plus"
              />
              <span
                class="dropdown--selected"
              >
                Create Task
              </span>
              <span
                class="dropdown--caret icon caret-down"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "store": Object {
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
    Symbol(observable): [Function],
  },
  "unmount": [Function],
}
`;
